tests/getRoomById.js

```diff
@@ -0,0 +1,61 @@
+
+/**
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+const expect = require('chai').expect;
+const config = { refocusUrl: 'zzz', token: 'dummy' };
+// eslint-disable-next-line max-len
+const bdk = require('../refocus-bdk-server')(config);
+const generic = require('../generic');
+const sinon = require('sinon');
+
+const room = {
+  'body': {
+    'id': 2,
+    'name': 'testRoom',
+    'origin': 'other',
+    'active': false,
+    'bots': [
+      'test-Bot'
+    ],
+    'type': 'e118e6f1-a6d5-4164-9c14-7344c733c3e6'
+  }
+};
+
+describe('getRoomById', () => {
+  let getGenericRoomStub;
+  beforeEach(() => {
+    getGenericRoomStub = sinon.stub(generic, 'get');
+  });
+
+  afterEach(() => {
+    getGenericRoomStub.restore();
+  });
+
+  it('OK if roomTypeId matches response', async () => {
+    getGenericRoomStub.returns(Promise.resolve(room));
+    const roomId = 2;
+    const roomBody = await bdk.getRoomById(roomId);
+    expect(roomBody.id).to.be.equal(roomId);
+  });
+
+  it('OK should fail when invalid id', async () => {
+    getGenericRoomStub.returns(Promise.resolve(room));
+    const roomBody = await bdk.getRoomById();
+    // eslint-disable-next-line no-unused-expressions
+    expect(roomBody).to.be.null;
+  });
+
+  it('OK should be null in when request fails', async () => {
+    getGenericRoomStub.returns(Promise.resolve());
```

Comment:
is this making the flow going to the catch?

Reply:
yes, this causes an undefined from the get (bad response) and is handled by the catch returning nulll.